src/app/forms/validators/custom-async-validators.ts

```diff
@@ -82,4 +82,31 @@ export class CustomAsyncValidators {
       );
     };
   }
-}
+
+  static testAndSwitchToHiddenFieldWithDefectTaxonomy(store: Store<State>): AsyncValidatorFn {
```

Comment:
Would it not be better to have this templated? If tests use defects taxonomy, then just make the result hidden, if they don't make it radiobuttons. Seems like this implementation somewhat defeats the purpose of templates

Reply:
That has been done - the async validators have been removed and the group test section templates have been changed to specify if the test result field is hidden or a radio button.